doc/ADR/homebrew_tap.md

```diff
@@ -0,0 +1,105 @@
+# Homebrew tap instructions
+
+CBMC archives versions into a tap which you can use to quickly download and
+build various historical versions.
+
+The tap is available at https://github.com/diffblue/homebrew-cbmc.
+
+Any version installed from the tap is going to be built from source.
+
+## User instructions
+
+1. Add the CBMC tap using the following command:
+
+   ```sh
+   $ brew tap diffblue/cbmc
+   ```
+
+2. Install any of the available CBMC versions in the following manner:
+
+   ```sh
+   $ brew install diffblue/cbmc/cbmc@5.54.0
+   ```
+
+3. To remove the version of CBMC installed you need to issue
+
+   ```sh
+   $ brew remove diffblue/cbmc/cbmc@5.54.0
+   ```
+
+4. To remove the tap altogether, you need to issue
+
+   ```sh
+   $ brew untap diffblue/cbmc
+   ```
+
+## Developer instructions
+
+1. First of all, you need to create a new tap locally:
+
+   ```sh
+   $ brew tap-new diffblue/homebrew-cbmc
+   ```
+
+   This step is needed if you are the one creating the tap.
+   If you want to install the tap that's already available
+   at the [Diffblue repository](https://github.com/diffblue/homebrew-cbmc),
+   `$ brew tap diffblue/cbmc` should suffice.
+
+
+2. For any edit, the second step is to `cd` into the repository.
+   To do that easily, it's best to ask Homebrew to give you the
+   location of the repository:
+
+   ```sh
+   $ cd $(brew --repo diffblue/cbmc)
+   ```
+
+3. You can extract versions of CBMC previously available in `homebrew/core` 
+   to the tap with the following instruction:
+
+   ```sh
+   $ brew extract cbmc diffblue/cbmc --version=<ver>
+   ```
+
+   For example `$ brew extract cbmc diffblue/cbmc --version=5.54.0` will
+   have as an end result a new file under `Formula/cbmc@5.54.0.rb` that will
+   contain the formula that we had submitted for that version of CBMC in the
+   `homebrew/core` repository.
+
+4. If you want to edit a specific version formula, you can do that by issueing
+
+   ```sh
+   $ brew edit cbmc@5.54.0
+   ```
+
+   with the appropriate version tag. This will open the formula on the default
+   editor in your system to allow you to make changes.
+
+   These changes are only going to reflected locally - but you can `cd` to the
+   repository, commit and then push to the tap (this is allowed only for people
+   who have permission to push to the tap - mainly, the Diffblue open source team).
+
+
+## Notes
+
+* Bootstrapping the repository with `brew tap --new` created two github actions
+  `.github/workflows/publish.yml` and `.github/workflows/tests.yml` which we are
+  not using at the moment but could serve to make future automation easier.
+
+## Known Limitations
+
+* The binaries are installed without any version tag attached, so they cannot
+  live side by side.
+* If you install a different version of CBMC and you want to use that one
+  instead, you need to take an extra step after installation to setup the
+  appropriate symlinks by using the `brew link` instruction:
```

Comment:
Where are these symlinks placed, what are they named and where do they link to?

Reply:
The symbolic links are in Homebrew's installation directory (default appears to be `/opt/homebrew/bin`.

The reason you need that is that when you install a CBMC version from the tap, the binaries end up in a folder like:

```
/opt/homebrew/Cellar/cbmc@5.55.0/5.55.0/bin/
```

That is not in the `$PATH` though - but `/opt/homebrew/bin` is.

This is why it automatically sets up a symlink to the installation that's linked:

```
/opt/homebrew/bin/cbmc -> /opt/homebrew/Cellar/cbmc@5.55.0/5.55.0/bin/cbmc
```

It does that for all binaries (`goto-cc`, `goto-analyzer`, etc).

This is why the extra linking step is needed.